src/Persistent/Implementation/AcidState.hs

```diff
@@ -29,23 +31,39 @@ import Persistent.Api
 
 
 mkRunPersistGeneric :: String
-                    -> (AulaData -> IO (AcidState AulaData))
-                    -> (AcidState AulaData -> IO ())
+                    -> (AulaData -> IO (AcidState AulaData, a))
+                    -> (AcidState AulaData -> a -> IO ())
                     -> IO RunPersist
 mkRunPersistGeneric desc openState closeState = do
-    st <- openState emptyAulaData
+    (st, handle) <- openState emptyAulaData
     pure RunPersist { _rpDesc   = desc
                     , _rpQuery  = query st AskDb
                     , _rpUpdate = update st
-                    , _rpClose  = closeState st
+                    , _rpClose  = closeState st handle
                     }
 
 mkRunPersistOnDisk :: Config -> IO RunPersist
 mkRunPersistOnDisk cfg =
-    mkRunPersistGeneric "acid-state (disk)"
-        (openLocalStateFrom $ cfg ^. dbPath) createCheckpointAndClose
+    mkRunPersistGeneric "acid-state (disk)" opn cls
+  where
+    opn aulaData = do
+        st <- openLocalStateFrom (cfg ^. persistConfig . dbPath) aulaData
+        tid <- forkIO . forever $ do
```

Comment:
Please add exception handling, and restart the thread if something terrible happened. Also some log entry would be fine here, about the exception.